### DISCUSSION TITLE
Themeing individual nodes

### DISCUSSION THREAD

**Discussion by aughey** [chrisjpatty/flume#165] [Ideas]
I'm looking to color nodes based on their type.  So logic nodes would be blue, action nodes red, etc.  Currently, it doesn't look like there is a way to do this.  Nodes in general can be themed through data-flume-component="node", however not based on some sort of configurable type.

To facilitate this, I'm testing in my fork adding a data-flume-component-type and data-flume-component-style to the top level Node attributes.  The type -type attribute will allow for a common case of coloring nodes based on their type.  For the -style attribute, I'm setting that to a {nodeStyle} attribute on the node called nodeStyle (attribute added in the nodesReducer).  

The final nodeStyle change is designed to allow for run-time visual styles to be added per-node.  In my case, I might color nodes based on their most recent execution execution to indicate which nodes are executing or which nodes might be disabled or have errors.

Thoughts?

**Comment by Perpaterb** [comment]
Super interested in this. 

If the nodes "type" and "id" could be added to the div class on creation that would make life easy. 

The same could be done for ports also.
